api/cluster/resource/templater_test.go

```diff
@@ -714,6 +726,127 @@ func TestCreateInferenceServiceSpec(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "pyfunc spec with model observability enabled; there is no effect ",
```

Comment:
If a user tries to enable model observability with `ModelTypePyFunc` or even the std model types for now, it would be better to let the API return an error, right? Shall we update `createPredictorSpec` function to return an error and propagate that?

Reply:
yes, it better to throw error, I made change to validation happen in the API layer so it can return early.